src/Symfony/Component/Workflow/DefinitionBuilder.php

```diff
@@ -122,6 +126,16 @@ public function addTransition(Transition $transition)
         return $this;
     }
 
+    /**
+     * @return $this
```

Comment:
I would use a type hint instead of a php doc

Reply:
yes ;)

Reply:
Suppose `NewDefinitionBuilder` extends our non-final `DefinitionBuilder`. Then return type of `$newDefinitionBuilder->setMetadataStore()` will be `DefinitionBuilder`, not `NewDefinitionBuilder`.

Reply:
Yes indeed. And it's not an issue ;)

Reply:
If `NewDefinitionBuilder` has `newMethod()` it won't be accessible in the chain of calls after `$newDefinitionBuilder->setMetadataStore()`, so fluent pattern will be broken.

Reply:
We do that everywhere in Symfony
More over: https://3v4l.org/v0cJj

Reply:
found some relevant occurrences in symfony code: 
![image](https://user-images.githubusercontent.com/2552865/39810598-3fce536e-538e-11e8-840c-bdc2a84a5a20.png)
but I still don't agree it is correct...

Reply:
With `: self` it is actually correct to do this:

```php
<?php

class A
{
    public function foo(): self
    {
        return $this;
    }
}

class B extends A
{
    public function foo(): A
    {
        return new A();
    }
}
```
It does not contradict the contract.

While `@return $this` indicates that the returned value should be the same instance.

Reply:
Makes sense to keep it as this provides more than just the return type.
In the code base, we use both return static and $this, dunno if there is a difference thought here